independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/AbstractSharedContext.java

```diff
@@ -87,7 +84,11 @@ public void destroy(Contextual<?> contextual) {
 
     @Override
     public synchronized void destroy() {
-        Set<ContextInstanceHandle<?>> values = instances.getAllPresent();
+        List<ContextInstanceHandle<?>> values = new LinkedList<>();
```

Comment:
I would not use the new method here. Note that shared contexts are only destroyed during app shutdown so I don't think it's a perf problem. 

BTW I don't know if it was @Sanne but he was very sceptical about the perf characteristics of `LinkedList` so we removed it from many places.

Reply:
I have used it for 3 reasons:
1. at https://github.com/quarkusio/quarkus/blob/98b0e6509fcad0dde90263cf8c3f9c163dfee8cf/independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ComputingCache.java#L98 
quoting the javadoc of `toSet` :
> There are no guarantees on the type, mutability, serializability, or thread-safety of the Set returned; if more control over the returned Set is required, use toCollection(Supplier).
which means that the `iterator::remove` on the iterator provided by the `Set` "could" (it is not, but is an impl detail we shouldn't rely on) break at runtime, because the `Set` it belongs is no longer mutable
2. it allow to piggyback on the existing test code which was hitting there
3. `LinkedList` are mem intensive and slower unless modification(s) has to happen in place and if the number of nodes is not "too big": I can still use an `ArrayList` here, nullying out removed elements, but I still don't know the required capacity upfront. In any case, the original code was allocating a set with a map within and an iterator, so that shouldn't be a problem

Reply:
For fun I created a O(1) array list removal too at https://github.com/franz1981/quarkus/commit/b8e88ddcdc41fa4a2c81b6d2fdc8b0c15974b5fc but TBH I think this is just too far and complex for a non-hot path